inline-lexer-test.js

```diff
@@ -293,3 +293,54 @@ test([
     ' ', 'STOP',
     ' ', 'STOP'
 ]);
+
+test(['! var="a \'hello\' variable"'], [
+  ' ',
+  'START',
+  '!',
+  'var',
+  'SYMBOL',
+  '=',
+  'STRING',
+  'a \'hello\' variable',
+  ' ',
+  'STOP',
+  ' ',
+  'STOP',
+  ' ',
+  'STOP'
+]);
+
+test(['! var=\'a hello \\nvariable\''], [
+  ' ',
+  'START',
+  '!',
+  'var',
+  'SYMBOL',
+  '=',
+  'STRING',
+  'a hello \nvariable',
+  ' ',
+  'STOP',
+  ' ',
+  'STOP',
+  ' ',
+  'STOP'
+]);
+
+test(['! var=\'a \\\'hello\\\' variable\''], [
+  ' ',
+  'START',
+  '!',
+  'var',
+  'SYMBOL',
+  '=',
+  'STRING',
+  'a \'hello\' variable',
+  ' ',
+  'STOP',
+  ' ',
+  'STOP',
+  ' ',
+  'STOP'
+]);
```

Comment:
Great tests.

Reply:
Can we also test multi-line `!` blocks to ensure the scanner works for strings that are not on the same line as the leader?

```
! Foo = 'Hello,'
  Bar = 'World!'
```